src/lib/logical_query_plan/functional_dependency.cpp

```diff
@@ -50,4 +52,43 @@ std::ostream& operator<<(std::ostream& stream, const FunctionalDependency& expre
   return stream;
 }
 
+std::vector<FunctionalDependency> merge_fds(const std::vector<FunctionalDependency>& fds_a,
+                                            const std::vector<FunctionalDependency>& fds_b) {
+  if constexpr (HYRISE_DEBUG) {
+    auto fds_a_set = std::unordered_set<FunctionalDependency>(fds_a.begin(), fds_a.end());
+    auto fds_b_set = std::unordered_set<FunctionalDependency>(fds_b.begin(), fds_b.end());
+    Assert(fds_a.size() == fds_a_set.size() && fds_b.size() == fds_b_set.size(),
+           "Did not expect input vector to contain multiple FDs with the same determinant expressions");
+  }
+  if (fds_a.empty() && fds_b.empty()) return {};
+  if (fds_a.empty()) return fds_b;
+  if (fds_b.empty()) return fds_a;
+
+  auto fds_merged = std::vector<FunctionalDependency>();
+  fds_merged.reserve(fds_a.size() + fds_b.size());
+  fds_merged.insert(fds_merged.end(), fds_a.begin(), fds_a.end());
+
+  for (const auto& fd_to_add : fds_b) {
+    auto existing_fd = std::find_if(fds_merged.begin(), fds_merged.end(),
+                                    [&fd_to_add](auto& fd) { return fd.determinants == fd_to_add.determinants; });
+    if (existing_fd == fds_merged.end()) {
+      fds_merged.push_back(fd_to_add);
+    } else {
+      // An FD with the same determinant expressions already exists. Therefore, we only have to add to the dependent
+      // expressions set
+      existing_fd->dependents.insert(fd_to_add.dependents.begin(), fd_to_add.dependents.end());
+    }
+  }
```

Comment:
The merge logic for FDs with duplicate determinant expressions seems necessary. But to be honest, I could not come up with an LQP scenario that requires the merging of such FDs. 
Should we stay with the above logic or replace it with pure vector-copying and an Assert?

Reply:
Not sure; I can't come up with anything right now either. Your call. If it stays, it should be tested though.

Reply:
Alright, I added one in JoinNodeTest and called it FunctionalDependenciesMerge.